spinoso-symbol/src/lib.rs

```diff
@@ -0,0 +1,370 @@
+#![warn(clippy::all)]
+#![warn(clippy::pedantic)]
+#![warn(clippy::cargo)]
+#![allow(unknown_lints)]
+#![warn(broken_intra_doc_links)]
+#![warn(missing_docs)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_copy_implementations)]
+#![warn(rust_2018_idioms)]
+#![warn(trivial_casts, trivial_numeric_casts)]
+#![warn(unused_qualifications)]
+#![warn(variant_size_differences)]
+#![forbid(unsafe_code)]
+// Enable feature callouts in generated documentation:
+// https://doc.rust-lang.org/beta/unstable-book/language-features/doc-cfg.html
+//
+// This approach is borrowed from tokio.
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![cfg_attr(docsrs, feature(doc_alias))]
+
+//! Identifier for interned bytestrings and routines for manipulating the
+//! underlying bytestrings.
+//!
+//! `Symbol` is a `Copy` type based on `u32`. `Symbol` is cheap to copy, store,
+//! and compare. It is suitable for representing indexes into a string interner.
+//!
+//! # Artichoke integration
```

Comment:
This is the main experiment with this PR. I think it is successful.

For a long time I was convinced with would require existential types. In practices, the `artichoke-core` traits are sufficiently granular and need to interact with the interpreter minimal enough for APIs like these. Manually specifying the trait bounds is not too terrible.